homeassistant/components/sensor/zamg.py

```diff
@@ -0,0 +1,273 @@
+    11126   Patscherkofel
+    11130   Kufstein
+    11150   Salzburg
+    11155   Feuerkogel
+    11157   Aigen im Ennstal
+    11171   Mariazell
+    11190   Eisenstadt
+    11204   Lienz
+"""
+
+import csv
+from datetime import timedelta
+import logging
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.weather import (
+    WeatherEntity, PLATFORM_SCHEMA, ATTR_WEATHER_HUMIDITY,
+    ATTR_WEATHER_PRESSURE, ATTR_WEATHER_TEMPERATURE,
+    ATTR_WEATHER_WIND_BEARING, ATTR_WEATHER_WIND_SPEED,
+    ATTR_WEATHER_ATTRIBUTION
+)
+from homeassistant.const import (
+    TEMP_CELSIUS, CONF_MONITORED_CONDITIONS,
+    CONF_NAME, STATE_UNKNOWN
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'zamg'
+ATTRIBUTION = 'Data provided by ZAMG'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)  # Only updates once per hour
+
+CONF_STATION_ID = "station_id"
+
+SENSOR_TYPES = {
+    ATTR_WEATHER_PRESSURE: ('Pressure', 'hPa', 'LDstat hPa', float),
+    'pressure_sealevel': ('Pressure at Sea Level', 'hPa', 'LDred hPa', float),
+    ATTR_WEATHER_HUMIDITY: ('Humidity', '%', 'RF %', int),
+    ATTR_WEATHER_WIND_SPEED: ('Wind Speed', 'km/h', 'WG km/h', float),
+    ATTR_WEATHER_WIND_BEARING: ('Wind Bearing', '°', 'WR °', int),
+    'wind_max_speed': ('Top Wind Speed', 'km/h', 'WSG km/h', float),
+    'wind_max_bearing': ('Top Wind Bearing', '°', 'WSR °', int),
+    'sun_last_hour': ('Sun Last Hour', '%', 'SO %', int),
+    ATTR_WEATHER_TEMPERATURE: ('Temperature', '°C', 'T °C', float),
+    'precipitation': ('Precipitation', 'l/m²', 'N l/m²', int),
+    'dewpoint': ('Dew Point', '°C', 'TP °C', float),
+    # The following probably not useful for general consumption,
+    # but we need them to fill in internal attributes
+    'station_name': ('Station Name', None, 'Name', str),
+    'station_elevation': ('Station Elevation', 'm', 'Höhe m', int),
+    'update_date': ('Update Date', None, 'Datum', str),
+    'update_time': ('Update Time', None, 'Zeit', str),
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+# ------------------------------------------------------------------------
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    "setup platform"
+
+    station_id = config.get(CONF_STATION_ID)
+    name = config.get(CONF_NAME)
+
+    probe = ZamgData(station_id=station_id, logger=_LOGGER)
+
+    sensors = [ZAMGWeather(probe, variable, name)
+               for variable in config[CONF_MONITORED_CONDITIONS]]
+
+    add_devices(sensors, True)
+
+
+# ------------------------------------------------------------------------
+class ZAMGWeather(WeatherEntity):
```

Comment:
You need to decide if you want to create a platform for the `weather` component or a "classical" sensor.